File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var rearrangeArray = function(nums) {
    const pos=[];
    const neg=[];
    const result=[];
    for(let i in nums){
        if(nums[i]>=0) pos.push(nums[i]);
        else neg.push(nums[i]);
    }
    for(let i in pos){
        result.push(pos[i]);
        result.push(neg[i]);
    }
    return result;
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int[] result = new int[nums.length];
        int p = 0, n = 1;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] >= 0) {
                result[p] = nums[i];
                p += 2;
            } else {
                result[n] = nums[i];
                n += 2;
            }
        }
        return result;
    }

}